include/seqan3/alphabet/composition.hpp

```diff
@@ -0,0 +1,211 @@
+#include <seqan3/alphabet/quality/concept.hpp>
+#include <seqan3/core/pod_tuple.hpp>
+#include <seqan3/core/detail/int_types.hpp>
+
+/*!\file alphabet/composition.hpp
+ * \ingroup alphabet
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+
+ * \brief Contains alphabet_composition.
+ */
+
+namespace seqan3
+{
+
+/*!\brief The CRTP base of alphabets that contain multiple (different) letters at one position.
+ * \ingroup alphabet
+ * \tparam first_alphabet_type Type of the first letter; must satisfy alphabet_concept.
+ * \tparam alphabet_types Types of further letters (up to 4); must satisfy alphabet_concept.
+ *
+ * This data structure is CRTP base clasee for combined alphabets, where the different
+ * alphabet letters exist independently, similar to a tuple. In fact this class
+ * provides a tuple-like interface with `std::get<0>(t)` and objects can be brace-initialized
+ * with the individual members.
+ *
+ * \attention
+ * This a "pure base class", you cannot instantiate it, you can only inherit from it.
+ * Most likely you are interested in using one of it's descendents like quality_composition.
+ *
+ * \sa quality_composition
+ * \sa mask_composition
+ */
+
+template <typename derived_type,
+          typename first_alphabet_type,
+          typename ...alphabet_types>
+      requires alphabet_concept<first_alphabet_type> && (alphabet_concept<alphabet_types> && ...)
+struct alphabet_composition :
+    public pod_tuple<first_alphabet_type, alphabet_types...>
+{
+public:
+    //!\brief The type of value_size and `alphabet_size_v<alphabet_composition<...>>`
+    using integral_type = detail::min_viable_uint_t<alphabet_size_v<first_alphabet_type> *
+                                                    (alphabet_size_v<alphabet_types> * ...)>;
+
+    //!\brief The product of the sizes of the individual alphabets.
+    static constexpr integral_type value_size{alphabet_size_v<first_alphabet_type> *
+                                              (alphabet_size_v<alphabet_types> * ...)};
+
+    //!\cond DEV
+    //!\brief the cummulative alphabet size products (first, first*second, first*second*third...) are cached
+    static constexpr std::array<integral_type, sizeof...(alphabet_types)+1> _cummulative_alph_sizes
+    {
+        [] () constexpr
+        {
+            std::array<integral_type, sizeof...(alphabet_types)+1> ret{};
+            size_t count = 0;
+            meta::for_each(meta::list<first_alphabet_type, alphabet_types...>{}, [&] (auto && alph) constexpr
+            {
+                ret[count] = alphabet_size_v<std::decay_t<decltype(alph)>> * (count > 0 ? ret[count - 1] : 1);
+                ++count;
+            });
+
+            return std::move(ret);
+        }()
+    };
+
+    //!\brief An index sequence up to the number of contained letters.
+    static constexpr auto _positions = std::make_index_sequence<sizeof...(alphabet_types)>{};
+    //!\endcond
+
+
+    /*!\name Read functions
+     * \{
```

Comment:
It results in different bahaviour. If I give it all one line comments, than the following `\brief` will be pulled into the group instead of annotating the following function.